File: examples/governance_static_go/domain/requirement.go
package domain

type Requirement struct {
	Description string `json:"description"`
	Priority string `json:"priority"`
	Category string `json:"category"`
}

func NewRequirement(description string, priority string, category string) (Requirement, error) {
	v := Requirement{
		Description: description,
		Priority: priority,
		Category: category,
	}
	return v, nil
}

File: examples/governance/compliance_domain/lib/compliance_domain/governance_policy/requirement.rb
module ComplianceDomain
  class GovernancePolicy
    class Requirement
      attr_reader :description, :priority, :category

      def initialize(description:, priority:, category:)
        @description = description
        @priority = priority
        @category = category
        check_invariants!
        freeze
      end

      def ==(other)
        other.is_a?(self.class) &&
          description == other.description &&
          priority == other.priority &&
          category == other.category
      end
      alias eql? ==

      def hash
        [self.class, description, priority, category].hash
      end

      private

      def check_invariants!; end
    end
  end
end
